### PR TITLE
Fix: improve storage tool checks, disable default allocator check, fix mwcsys_executil

### PR DESCRIPTION
**Pull request by 678098**
Storage tool:

- remove manual memory management
- add preconditions checks
- list all member fields on construction
- make the code robust to NULL allocator

### CODE REVIEW

File: src/groups/mwc/mwcsys/mwcsys_executil.t.cpp [hunk 1]
```diff
@@ -130,7 +130,7 @@ static void test2_executeSystemFailure()
 
     PVV("Testing abnormal exit of the command");
     rc = mwcsys::ExecUtil::execute(&output,
-                                   "python -c 'import os,signal; "
+                                   "python3 -c 'import os,signal; "
```
Comment (678098) [inline src/groups/mwc/mwcsys/mwcsys_executil.t.cpp:133]: We cannot use implicit `python2` here, `python3` alias might work on most frequently known systems

File: src/applications/bmqstoragetool/m_bmqstoragetool_journalfileprocessor.t.cpp [hunk 2]
```diff
@@ -758,6 +758,12 @@ static void test12_printMessagesDetailsTest()
 {
     mwctst::TestHelper::printTestName("PRINT MESSAGE DETAILS TEST");
 
+#if defined(BSLS_PLATFORM_OS_SOLARIS) || defined(BSLS_PLATFORM_OS_AIX)
+    s_ignoreCheckDefAlloc = true;
+    // Disable default allocator check for this test until we can debug
+    // it on AIX/Solaris
+#endif
```
Comment (678098) [inline src/applications/bmqstoragetool/m_bmqstoragetool_journalfileprocessor.t.cpp:765]: I am not sure what code do allocations wrongly, it is difficult to find without means to build and test it on AIX/Solaris. I reviewed the storage tool and improved many things related to allocations, but maybe we should just disable this check until we debug it on the needed platforms

Comment (pniedzielski) [reply src/applications/bmqstoragetool/m_bmqstoragetool_journalfileprocessor.t.cpp:765]: This makes sense to me.  It's at least better to do this test by default on Linux, which will give us at least some early indications with GH Actions.

File: src/groups/mwc/mwcsys/mwcsys_executil.t.cpp (hunk 1 above)
Comment (pniedzielski) [reply src/groups/mwc/mwcsys/mwcsys_executil.t.cpp:133]: I'm fine with this test.  If it fails, we can use use `python3.11` etc, but I'd rather not tie ourselves to a specific Python version if we can help it.

File: src/applications/bmqstoragetool/m_bmqstoragetool_testutils.h [hunk 3]
```diff
@@ -102,8 +102,8 @@ class JournalFile {
     // Number of records.
     MappedFileDescriptor d_mfd;
     // Mapped file descriptor.
-    char* d_mem_p;
-    // Pointer to allocated memory for journal file.
+    bsl::vector<char> d_mem;
```
Comment (pniedzielski) [inline src/applications/bmqstoragetool/m_bmqstoragetool_testutils.h]: Vector is fine (exception safe!).  I'm surprised there isn't a clear BDE component for this use case, but I couldn't find one with a quick search.

**Review by pniedzielski — APPROVED** [review]
👍🏻 as long as CI passes

File: src/applications/bmqstoragetool/m_bmqstoragetool_testutils.h (hunk 3 above)
Comment (678098) [reply src/applications/bmqstoragetool/m_bmqstoragetool_testutils.h]: I believe allocating data buffers via vector is a common practice for safe (not a joke, but still sounds funny) C++ programming, just as using smart pointers and other RAII things

Comment (678098) [reply src/applications/bmqstoragetool/m_bmqstoragetool_testutils.h]: From the past days, we used `levmar` lib for optimizations in C++ code, with API like this: 
```
int dlevmar_dif(
void (*func)(double *p, double *hx, int m, int n, void *adata)
```
And we were really happy to use vector buffers on top of this

**Pull request merged**